flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/BufferPoolFactory.java

```diff
@@ -49,10 +50,10 @@
 	 * 		minimum number of network buffers in this pool
 	 * @param maxUsedBuffers
 	 * 		maximum number of network buffers this pool offers
-	 * 	@param owner
-	 * 	    the optional owner of this buffer pool to release memory when needed
+	 * 	@param bufferPoolOwner
+	 * 	    the owner of this buffer pool to release memory when needed
 	 */
-	BufferPool createBufferPool(int numRequiredBuffers, int maxUsedBuffers, Optional<BufferPoolOwner> owner) throws IOException;
+	BufferPool createBufferPool(int numRequiredBuffers, int maxUsedBuffers, @Nullable BufferPoolOwner bufferPoolOwner) throws IOException;
```

Comment:
> Instead of @Nullable you can just use overloaded method without the special parameter.

We could remove `@Nullable` from this interface method, but it is still needed in specific implementation of `NetworkBufferPool#createBufferPool(int numRequiredBuffers, int maxUsedBuffers, @Nullable BufferPoolOwner bufferPoolOwner)`, because `NetworkBufferPool#createBufferPool(int numRequiredBuffers, int maxUsedBuffers)` would reuse the  above method to pass `null`. Or I did not get the point?

> We should avoid @Nullable parameters/return values. Having @Nullable private field is a lesser evil, as it's encapsulated by the current class and the nullability is not leaking to the interface/class users.

Totally agree with this general idea.

Reply:
I think the way to avoid this `@Nullable` on public/protected method in this case, is to have 
```
public BufferPool createBufferPool(int numRequiredBuffers, int maxUsedBuffers) {
  return createBufferPool(numRequiredBuffers, maxUsedBuffers, null);
}

public BufferPool createBufferPool(int numRequiredBuffers, int maxUsedBuffers, @NonNull owner) {
  return createBufferPoolInternal(numRequiredBuffers, maxUsedBuffers, owner);
}
```
and internal helper method:
```
private BufferPool createBufferPoolInternal(int numRequiredBuffers, int maxUsedBuffers, @Nullable owner)
```

But that's quite some overhead. This code bloat is one of the reasons why I don't mind `Optional` as a parameter :( 

LGTM regardless of the version that you pick.

Reply:
Thanks for the suggestion and I prefer this way a bit. It seems not very overhead. :)